### DISCUSSION TITLE
Best practices for saving the content in a database

### DISCUSSION THREAD

**Discussion by charklewis** [facebook/lexical#1840] [Q&A] [answered]
Are there any best practices on saving the data to a database? What data would we need to save, and how would we populate the composer with that data?

**Comment by zurfyx** [comment, marked as answer]
`JSON.stringify(editorState)` and `editor.setEditorState(editor.parseEditorState(editorStateJSON))`

You may want to check how file import/export works! https://github.com/facebook/lexical/blob/main/packages/lexical-file/src/fileImportExport.js

That said, and one of the reasons why we have a ⚠️ on the main README is that we're actively working on a versioning system, especially oriented towards custom app nodes.

Alternatively, you can build your own format based on the tree structure.

```
editor.getEditorState().read(() => {
  $getRoot() // iterate from here and build your own JSON or whatever format you prefer
});
```

The last works best when you already have documents in certain format on the database, like it is the case for many surfaces at Meta.

**Reply by charklewis** [reply]
Awesome, thanks! I'll have a look tomorrow. I have been really enjoying using the package so far.

**Reply by charklewis** [reply]
Do you think if I use `JSON.stringify(editorState)` there will be an upcoming breaking changes that would cause me any troubles?

**Reply by zurfyx** [reply]
Unlikely on our end. You may want to read over the first chunk even if it's just a draft - https://github.com/facebook/lexical/issues/1768

**Reply by mrkchang** [reply]
Is there a best practice on large corpus of text? e.g. a book of 300 pgs
